examples/lock-app/efr32/src/LockManager.cpp

```diff
@@ -407,6 +445,165 @@ bool LockManager::SetCredential(chip::EndpointId endpointId, uint16_t credential
+                                         EmberAfPluginDoorLockYearDaySchedule & schedule)
+{
+    if ((yearDayIndex < MIN_SCHEDULE_INDEX) || (yearDayIndex > MAX_SCHEDULE_PER_USER))
+    {
+        ChipLogError(Zcl, "Cannot set schedule - schedule index is out of range [endpoint=%d,index=%d]", endpointId, yearDayIndex);
+        return DlStatus::kFailure;
+    }
+    if ((userIndex < MIN_USER_INDEX) || (userIndex > DOOR_LOCK_MAX_USERS))
+    {
+        ChipLogError(Zcl, "Cannot get schedule - user index is out of range [endpoint=%d,index=%d]", endpointId, userIndex);
+        return DlStatus::kFailure;
+    }
+
+    // doorlock server checks for valid indices
+    uint8_t adjustedYearDayIndex = yearDayIndex - 1;
+    uint16_t adjustedUserIndex   = userIndex - 1;
+
+    auto & scheduleInStorage = mYeardaySchedule[adjustedUserIndex][adjustedYearDayIndex];
+
+    schedule = scheduleInStorage;
+
+    return DlStatus::kSuccess;
+}
+
+DlStatus LockManager::SetYeardaySchedule(chip::EndpointId endpointId, uint8_t yearDayIndex, uint16_t userIndex,
+                                         DlScheduleStatus status, uint32_t localStartTime, uint32_t localEndTime)
+{
+    if ((yearDayIndex < MIN_SCHEDULE_INDEX) || (yearDayIndex > MAX_SCHEDULE_PER_USER))
+    {
+        ChipLogError(Zcl, "Cannot set schedule - schedule index is out of range [endpoint=%d,index=%d]", endpointId, yearDayIndex);
+        return DlStatus::kFailure;
+    }
+    if ((userIndex < MIN_USER_INDEX) || (userIndex > DOOR_LOCK_MAX_USERS))
+    {
+        ChipLogError(Zcl, "Cannot get schedule - user index is out of range [endpoint=%d,index=%d]", endpointId, userIndex);
+        return DlStatus::kFailure;
+    }
+
+    // doorlock server checks for valid indices
+    uint8_t adjustedYearDayIndex = yearDayIndex - 1;
+    uint16_t adjustedUserIndex   = userIndex - 1;
+
+    auto & scheduleInStorage = mYeardaySchedule[adjustedUserIndex][adjustedYearDayIndex];
+
+    // scheduleInStorage.status = status;
+    scheduleInStorage.localStartTime = localStartTime;
+    scheduleInStorage.localEndTime   = localEndTime;
+
+    EFR32Config::WriteConfigValueBin(EFR32Config::kConfigKey_YearDaySchedules, reinterpret_cast<const uint8_t *>(mYeardaySchedule),
+                                     sizeof(EmberAfPluginDoorLockYearDaySchedule) * mNumberOfYeardaySchedulesPerUserSupported *
+                                         mMaxUsers);
+
+    return DlStatus::kSuccess;
+}
+
+DlStatus LockManager::GetHolidaySchedule(chip::EndpointId endpointId, uint8_t holidayIndex,
+                                         EmberAfPluginDoorLockHolidaySchedule & schedule)
+{
+    if ((holidayIndex < MIN_SCHEDULE_INDEX) || (holidayIndex > MAX_SCHEDULE_PER_USER))
+    {
+        ChipLogError(Zcl, "Cannot set schedule - schedule index is out of range [endpoint=%d,index=%d]", endpointId, holidayIndex);
+        return DlStatus::kFailure;
+    }
+
+    // door-lock server checks for valid indices
```

Comment:
what does this comment mean? Maybe `valid index validated above` ?

Reply:
Comment was removed along with adjustedHolidayIndex